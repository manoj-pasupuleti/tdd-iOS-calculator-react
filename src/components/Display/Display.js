import React from "react";
import PropTypes from "prop-types";

import "./Display.css";

export default function Display({ displayValue }) {
  return (
    <div className="display-container">
      <p className="display-value">{displayValue}</p>
    </div>
  );
}

Display.propTypes = {
  displayValue: PropTypes.string.isRequired
};
