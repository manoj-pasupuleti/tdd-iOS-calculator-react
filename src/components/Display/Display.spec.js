import React from "react";
import { shallow } from "enzyme";
import Display from "./Display";
import { exportAllDeclaration } from "@babel/types";

const TEST_DISPLAY_VALUE = "Testing123";

describe("Display", () => {
  let wrapper;
  beforeEach(() => {
    wrapper = shallow(<Display displayValue={TEST_DISPLAY_VALUE} />);
  });

  it("Should Render Correctly", () => {
    expect(wrapper).toMatchSnapshot();
  });

  it("Renders display value correctly", () => {
    expect(wrapper.text()).toEqual(TEST_DISPLAY_VALUE);
  });
});
