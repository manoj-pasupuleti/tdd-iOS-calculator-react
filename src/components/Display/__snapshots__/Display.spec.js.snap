// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Display Should Render Correctly 1`] = `
<div
  className="display-container"
>
  <p
    className="display-value"
  >
    Testing123
  </p>
</div>
`;
